images/cfw/opt/x1plus/lib/python/x1plus/services/x1plusd/modules/actions/delay.py

```diff
@@ -0,0 +1,18 @@
+"""
+[module]
+enabled=true
+[end]
+"""
+from x1plus.services.x1plusd.actions import register_action
+
+import asyncio
+import logging
+
+logger = logging.getLogger(__name__)
+
+@register_action("delay")
+async def _action_delay(handler, subconfig):
+    logger.debug(f"delay action: {subconfig}")
+    if type(subconfig) != int and type(subconfig) != float:
+        raise TypeError(f"delay parameter {subconfig} was not numberish")
+    await asyncio.sleep(subconfig)
```

Comment:
my only thought on this is that it is a good example, but I really would prefer to have core behavior not be modular, lest there be users forgetting they disabled something and then wondering why it does not work

Reply:
I'm 50/50 on it - if a user disabled something like that, it would be a very obvious insert in their `x1plus settings` and be very hidden. And standardization has its benefits especially since it's not wildly unique, hence the compromise I had where this is modular, but it's in the core directories and is always enabled by default.

But I could be convinced if we kept say, some single one modular as an example, and the rest we re-merged into a single file within the `actions.py`. If it's meant to be extremely core, then it doesn't necessarily need to be structured modularly either, though it is very simple/clean to so yeah, 50/50 on it.